multinet/api.py

```diff
@@ -112,7 +138,29 @@ def get_node_edges(
 @swag_from("swagger/create_workspace.yaml")
 def create_workspace(workspace: str) -> Any:
     """Create a new workspace."""
-    db.create_workspace(workspace)
+
+    # Set up a new ArangoDB document to describe the newly created workspace.
+    new_doc: Workspace = {
+        "name": workspace,
+        "internal": util.generate_arango_workspace_name(),
+        "permissions": {
+            "owner": "",
+            "maintainers": [],
+            "writers": [],
+            "readers": [],
+            "public": False,
+        },
+    }
+
+    # Assign owner and public flags depending on who's logged in.
+    user = session_user()
+    if user is None:
+        new_doc["permissions"]["public"] = True
+    else:
+        new_doc["permissions"]["owner"] = user.sub
```

Comment:
This means that users that are not logged in can make workspaces. Is this what we resolved to do? Also, @waxlamp, you mentioned that you could make the owner an optional string, instead of just blank, which sounds like a good idea

Reply:
> This means that users that are not logged in can make workspaces. Is this what we resolved to do?

This is exactly what we need to discuss; see the PR description.

> Also, @waxlamp, you mentioned that you could make the owner an optional string, instead of just blank, which sounds like a good idea

That does make more sense from a typing standpoint, but from an execution standpoint, there's a nice advantage to using a blank string for "no owner": no real user has a blank string as their `sub` value, so it leaves all the code for processing users nice and simple. OTOH, the type safety is pretty attractive. I will probably make this change but if there are opposing viewpoints, now would be a good time to hear them.

Reply:
> > This means that users that are not logged in can make workspaces. Is this what we resolved to do?
> 
> This is exactly what we need to discuss; see the PR description.

Understood. I think we shouldn't allow users that are not logged in to create workspaces. This could lead to abuse, eventually, and it makes sense that every workspace should have an owner. A workspace not having an owner could lead to confusion, where users make workspaces when they're not logged in, then they become public and they're unable to edit them.

I welcome any counter arguments though

> > Also, @waxlamp, you mentioned that you could make the owner an optional string, instead of just blank, which sounds like a good idea
> 
> That does make more sense from a typing standpoint, but from an execution standpoint, there's a nice advantage to using a blank string for "no owner": no real user has a blank string as their `sub` value, so it leaves all the code for processing users nice and simple. OTOH, the type safety is pretty attractive. I will probably make this change but if there are opposing viewpoints, now would be a good time to hear them.

I think we should make the change for better typing. I think making it an optional string will mean that we're less like to forget about the edge case where a workspace has no owner.

Reply:
> > > This means that users that are not logged in can make workspaces. Is this what we resolved to do?
> > 
> > 
> > This is exactly what we need to discuss; see the PR description.
> 
> Understood. I think we shouldn't allow users that are not logged in to create workspaces. This could lead to abuse, eventually, and it makes sense that every workspace should have an owner. A workspace not having an owner could lead to confusion, where users make workspaces when they're not logged in, then they become public and they're unable to edit them.
> 
> I welcome any counter arguments though
> 

I agree that we should just not allow it for unlogged-in users. @AlmightyYakob, any objection?

I will add a commit to handle this case; it'll be a good prototype use case for a `401 Unauthorized` response.

> > > Also, @waxlamp, you mentioned that you could make the owner an optional string, instead of just blank, which sounds like a good idea
> > 
> > 
> > That does make more sense from a typing standpoint, but from an execution standpoint, there's a nice advantage to using a blank string for "no owner": no real user has a blank string as their `sub` value, so it leaves all the code for processing users nice and simple. OTOH, the type safety is pretty attractive. I will probably make this change but if there are opposing viewpoints, now would be a good time to hear them.
> 
> I think we should make the change for better typing. I think making it an optional string will mean that we're less like to forget about the edge case where a workspace has no owner.

If we're not allowing ownerless workspaces, then the `owner` field *should* be `str` rather than `Optional[str]` no?

Reply:
> I agree that we should just not allow it for unlogged-in users. @AlmightyYakob, any objection?

@waxlamp I agree, only logged in user should be able to create a workspace. 

> If we're not allowing ownerless workspaces, then the `owner` field _should_ be `str` rather than `Optional[str]` no?

I think so, I don't see any particular benefit to making it optional. However, this means that we'll have to do some migration on existing workspaces. Maybe we could set them to some default user that we create (and have access to), and then we can assign them to whoever they belong to.

Reply:
I agree with Jake on that last point

Reply:
> Maybe we could set them to some default user that we create (and have access to), and then we can assign them to whoever they belong to.

Since we don't have an in-app admin console yet, I don't see the value in assigning these to an existing user, as opposed to just leaving the string blank, and setting the workspaces to be public. I wrote a migration script (2d1491c143d10d10b0b6a07bb26209f6590fc36d in this PR) to do just that. Once the actual owners of these workspaces have accounts, we can re-assign them manually.

Reply:
Added a 401 error for logged-out users trying to create a workspace: see dbabdbc.